src/ir/possible-constant.h

```diff
@@ -118,6 +105,22 @@ struct PossibleConstantValues {
       return true;
     }
 
+    // Nulls compare equal, and we could consider any of the input nulls as the
+    // combination of the two (as any of them would be valid to place in the
+    // location we are working to optimize). In order to have simple symmetric
+    // behavior here, which does not depend on the order of the inputs, use the
+    // LUB.
+    if (isNull() && other.isNull()) {
+      auto type = getConstantLiteral().type.getHeapType();
+      auto otherType = other.getConstantLiteral().type.getHeapType();
+      auto lub = HeapType::getLeastUpperBound(type, otherType);
```

Comment:
Can we directly use `Type::getLeastUpperBound`?

Reply:
We can, but I think the clearer thing is to LUB on the heap type, because nullability is not relevant here... but maybe the real issue here is that `makeNull` gets a type and not a heap type - if it got a heap type then we wouldn't have two options here, which seems better. Let me look into that as a refactor PR.

Reply:
Both of the types are null at this point. Doesn't that mean both types are nullable anyway here?

Reply:
Yes, exactly, the types must be nullable anyhow. So we really just need the least upper bound of the heap type, which is the only thing that can be different between them.

After #4664 the code here will be something like this, which I think is good:
```cpp
auto lub = HeapType::getLeastUpperBound(type, otherType)
if (lub != type) {
  value = Literal::makeNull(lub);
  return true;
}
```